src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -569,6 +587,12 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                         bytesRead = await vt.ConfigureAwait(false);
                     }
 
+                    if (_readAheadTaskException is not null)
+                    {
+                        // The pre-emptive read triggered by CheckUsabilityOnScavenge threw an exception, rethrow it here
+                        throw _readAheadTaskException;
```

Comment:
Should we null this out, e.g.
```C#
if (_readAheadTaskException is Exception e)
{
    _readAheadTaskException = null;
    throw e;
}
```
?  It may not matter in practice, since if we're throwing this connection shouldn't end up being reused.